xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAESink.cpp

```diff
@@ -837,6 +837,78 @@ void CActiveAESink::OpenSink()
   CLog::Log(LOGDEBUG, "CActiveAESink::OpenSink - trying to open device {}", device);
   m_sink = CAESinkFactory::Create(device, m_sinkFormat);
 
+  // try other device with same driver and same friendly name
+  // is probably the same device renamed due driver update
+  if (!m_sink && !m_sinkInfoList.empty() && !dev.driver.empty() && !dev.friendlyName.empty())
+  {
+    for (const auto& sink : m_sinkInfoList)
+    {
+      if (sink.m_sinkName != dev.driver)
+        continue;
+
+      if (m_sink)
+        break;
+
+      for (const auto& d : sink.m_deviceInfoList)
+      {
+        if (d.m_displayName == dev.friendlyName || d.m_displayNameExtra == dev.friendlyName)
+        {
+          m_deviceFriendlyName = dev.friendlyName;
```

Comment:
There is an existing GetDeviceFriendlyName(dev.name) a bit earlier in the function with different logic (uses m_displayName), which way is correct? Maybe it's different for Windows and other platforms.

Reply:
Yes for consistency better always use `GetDeviceFriendlyName` from deviceName